### ISSUE TITLE
Convert Preprocessing tests to lit

### ISSUE THREAD

**Issue by LukeIreland1** [flang-compiler/f18#1052]
Preprocessing tests need to be converted to run with lit, like the other Fortran tests.

**Comment by DavidTruby** [comment]
@RichBarton-Arm is this complete?

**Comment by RichBarton-Arm** [comment]
No, the tests in the test/Preprocessing directory are not ported to lit yet. These tests were never being run on F18 and we tell lit to ignore them right now. 

The tests are execution tests, i.e. they build Fortran snippets that run and print Pass/Fail messages. Each test relies on some functionality from the preprocessor being correctly honoured in order for the Pass message to print. 

It should not be necessary to compile and run the test in order to check the preprocessor. I would expect tests on the output of -E should be sufficient and more desirable because they isolate the preprocessor so would not fail due to any subsequent issue unrelated to the preprocessor. I think this should be done at the same time as making these lit tests so in reality this might be more than a port, more a re-write/consolidate and porting job.
